schoenrock-rossiter_tom_ndvi.py

```diff
@@ -0,0 +1,656 @@
+# 
+# Your submission to the GitHub repository should include:
+# * This Jupyter Notebook file (.ipynb) with:
+#     * The code to create a plot of mean NDVI across a year for  2 NEON Field Sites:
+#         * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object
+#     * The **data should be cleaned to remove the influence of clouds**. See the [earthdatascience website for an example of what your plot might look like with and without removal of clouds](https://www.earthdatascience.org/courses/earth-analytics-python/create-efficient-data-workflows/).
+# * BONUS: Create one output `.csv` file that has 3 columns - NDVI, Date and Site Name - with values for SJER and HARV.
+# 
+# Your notebook should:
+# * Have *at least* 2 well documented and well named functions with docstrings.
+# * Include a Markdown cell at the top of the notebook that outlines the overall workflow using pseudocode (i.e. plain language, not code)
+# * Include additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Replace this cell with your pseudocode  for this workflow
+# 
+# If you happen to be a diagram person a diagram is ok too
+# 
+# 
+
+# Workflow pseudocode
+# 
+# Run first for one month at one site - next run for all months in one site - last run for all months in both sites
+# 
+# Preparation - 
+# - Import packages
+# - Set working directory to 'data'
+# - Download data from earthpy
+# 
+# Access Landsat band files for single scene -
+# - Create path to Landsat data
+# - Create list of Landsat bands in scene (into df?)
+# - Subset bands into just RGB, NIR
+# - Stack bands into single raster
+# - clip stacked raster by boundary layer
+# - Output raster for individual month
+# 
+# Access Landsat band files for all scenes in a site -
+# - Get list of all directories
+# - For bands in each scene directory perform the above steps
+# - Output, into a list or df, the average ndvi and data for each scene
+# 
+# Access Landsat band files for all scenes in both sites -
+# 
+# Calculate ndvi -
+# - Run ndvi expression on red and NIR bands
+# - Calculate mean ndvi for site
+# - Output ndvi df/raster
+# - Output ndvi values to spreadsheet
+# 
+# Plot ndvi -
+# - Create ndvi graph
+# 
+
+# In[1]:
+
+
+# Autograding imports - do not modify this cell
+import matplotcheck.autograde as ag
+import matplotcheck.notebook as nb
+import matplotcheck.timeseries as ts
+from datetime import datetime
+
+
+# In[2]:
+
+
+# Import needed packages in PEP 8 order
+# and no unused imports listed (10 points total)
+
+import os
+from glob import glob
+
+import matplotlib.pyplot as plt
+from matplotlib.colors import ListedColormap
+from matplotlib import patches as mpatches
+from matplotlib import colors
+from matplotlib.dates import DateFormatter
+
+import seaborn as sns
```

Comment:
I personally would list sns below pandas and numpy, since these two are more wide-used. But it maybe personal preference, and I remember Elsa saying that there are various approached to oder list of imported libraries.